engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -65,15 +98,38 @@ public void resetCamera() {
     }
 
     @ReceiveEvent
-    public void resetCamera(ResetCameraEvent resetCameraEvent, EntityRef client, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
+    public void resetCameraForClient(ResetCameraEvent resetCameraEvent, EntityRef entityRef, ClientComponent clientComponent) {
+        if (localPlayer.getClientEntity().equals(entityRef)) {
+            clientComponent.camera.destroy();
+            clientComponent.camera = EntityRef.NULL;
+            // this will trigger a ClientComponent change which will in turn trigger recreation of the camera entity
+            localPlayer.getClientEntity().saveComponent(clientComponent);
+        }
+    }
+
+    @ReceiveEvent
+    public void mountCameraOnActivate(OnActivatedComponent event, EntityRef entityRef, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
+        if (localPlayer.getClientEntity().equals(entityRef) && clientComponent.camera.exists()) {
+            mountCamera();
+        }
+    }
+
+    @ReceiveEvent
+    public void mountCameraOnChange(OnChangedComponent event, EntityRef entityRef, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
```

Comment:
As I read the names of both _mountCameraOnActivate_ and _mountCameraOnChange_ methods I wonder: what is activated? What is changed? Can you clarify with verbose names or javadocs?

Also, the code inside both methods is identical. How about we shorten it by removing the if statement from both and by passing the condition in one or two arguments to mountCamera, i.e.:

mountCamera(localPlayer.getClientEntity().equals(entityRef)<sup>1</sup>, clientComponent.camera.exists());

We would then put the IF statement in mountCamera but in a more readable form. I.e. 

```java
private void mountCamera(boolean weAreDealingWithTheRightEntity, boolean aCameraAlreadyExist) {
    if(weAreDealingWithTheRightEntity && aCameraAlreadyExist) { // do the magic }
}
```

<sup>1</sup>I don't actually understand what this does.